### PR TITLE
[feature][k8s application]K8s auto deploy

### PR DESCRIPTION
**Pull request by zackyoungh**
## Purpose of the pull request
1. k8s docker auto build and push
image

2. support k8s application udf

## issue
#1343

### CODE REVIEW

File: dlink-admin/src/main/java/com/dlink/service/impl/TaskServiceImpl.java [hunk 1]
```diff
@@ -408,25 +408,27 @@ public Task getTaskInfoById(Integer id) {
...
-            if (CollUtil.isNotEmpty(task.getConfig()) && Asserts.isNullString(task.getStatement())) {
-                Map<String, String> config = task.getConfig().get(0);
-                UDFTemplate template = udfTemplateService.getById(config.get("templateId"));
-                if (template != null) {
-                    String code = UDFUtil.templateParse(task.getDialect(), template.getTemplateCode(),
-                        config.get("className"));
-                    task.setStatement(code);
+        if (CollUtil.isNotEmpty(task.getConfig())) {
+            if (Dialect.isUDF(task.getDialect()) && Convert.toInt(task.getConfig().get(0).get("templateId"), 0) != 0) {
+                if (CollUtil.isNotEmpty(task.getConfig()) && Asserts.isNullString(task.getStatement())) {
```
Comment (leechor) [inline dlink-admin/src/main/java/com/dlink/service/impl/TaskServiceImpl.java]: 这里判断跟外面的if重复了.

File: dlink-common/src/main/java/com/dlink/utils/DockerClientUtils.java [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+     *
+     * @param client
+     * @param containerId
+     */
+    public void startContainer(DockerClient client, String containerId) {
+        client.startContainerCmd(containerId).exec();
+    }
+
+    /**
+     * 启动容器
```
Comment (Zzm0809) [inline dlink-common/src/main/java/com/dlink/utils/DockerClientUtils.java]: 这里是停止容器

File: dlink-admin/src/main/java/com/dlink/service/impl/TaskServiceImpl.java (hunk 1 above)
Comment (zackyoungh) [reply dlink-admin/src/main/java/com/dlink/service/impl/TaskServiceImpl.java]: 已修改

File: dlink-common/src/main/java/com/dlink/utils/DockerClientUtils.java (hunk 2 above)
Comment (zackyoungh) [reply dlink-common/src/main/java/com/dlink/utils/DockerClientUtils.java]: 已修改

**Review by aiwenmo — APPROVED** [review]
LGTM

**Pull request merged**